core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -205,31 +193,12 @@ public void serverStarting(Server server) throws Exception {
                         config.serviceConfigs().stream()
                               .filter(se -> virtualHosts.contains(se.virtualHost()))
                               .collect(toImmutableList());
-
-                final DocStringSupport docStringSupport = new DocStringSupport(services);
-                ServiceSpecification spec = generate(services);
-                spec = docStringSupport.addDocStrings(spec);
-                spec = exampleSupport.addExamples(spec);
-
-                final List<Version> versions = ImmutableList.copyOf(
-                        Version.getAll(DocService.class.getClassLoader()).values());
-
-                vfs().put("/specification.json",
-                          jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsBytes(spec));
-                vfs().put("/versions.json",
-                          jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsBytes(versions));
+                vfs().specificationLoader.updateServices(services);
+                vfs().specificationLoader.triggerPreload(config.blockingTaskExecutor());
```

Comment:
I thought it wouldn't be an issue since the task is scheduled before any requests are received (since we use a `ScheduledExecutorService` by default), but I guess other tasks can be affected depending on the executor implementation. Fixed!